### PR TITLE
feat(plugin-contributors-details): Add plugin-contributors-details

### LINKED ISSUE #30: Add a plugin for displaying authors from AUTHORS
Given an AUTHORS file of the format

```
Flip <flipactual@gmail.com>

```

we can fetch user information from GitHub with queries like `https://api.github.com/search/users?q=flipactual@gmail.com+in:email`

---

This will list anyone in the AUTHORS file in order, which at first seems all "mission accomplished", but it would require curating entries in AUTHORS because you probably don't want 50+ contributors listed in a file like a README (could be very difficult to read)... 

And that's why I'm not prioritizing this work. I'm not opposed to it, I think this is a good approach, but I don't see it as a priority

### PR DESCRIPTION
**Pull request by infiniteluke**
# Contributors Details Plugin

Uses an AUTHORS file containing name/email to fetch contributor data from GitHub API. Generates a details tag which lists contributors image/name.
Closes #30

## Example
I used this plugin in the main emdaer readme and this was the output:
<details>
<summary><strong>Contributors</strong></summary><br />

  <strong>Flip</strong>
</img></br></br>

  <strong>Luke Herrington</strong>
</img></br></br>

  <strong>Elliott Foster</strong>
</img></br></br>

  <strong>David Diers</strong>
</img></br></br>
</details>

## Questions
- [x] How do we want to handle offline? Maybe we can create a `contributor-data.json` file on each successful request and use that as a fall back if we're offline? `preuninstall` script could delete the file.
> I ended up creating an `.offline` directory underneath `.emdaer` to hold this and any other things like this in the future.
- [x] Currently there is no way to show a warning in emdaer, I found myself wanting to do that for the above.
> I don't _need_ this for this feature, but it would be nice to say `Hey, since you don't have connection, I used a cached file for the contributors details`
- [x] Should we make fetching the users a shared utility across all future contributor plugins (table, list, etc...)
> It's really simple and could be different across plugins, so I don't think so.
- [x] This approach will not work for people who's emails are not public. For example, @thebruce could not be added to the list. Should we use name/username instead of email?
> I changed to using github logins `Luke <infiniteluke>` and now everyone's included.
- [x] node-fetch has a flow-typed definition, but it didn't work with async/await so I stubbed it. Cool?
> This isn't a big deal in my mind.
- [X] Document offline support in CONTRIBUTING

### Notes
- This needs tests, but I figured we can discuss the above and the approach before I spend time on that.

### CODE REVIEW

**Comment by infiniteluke** [comment]
doing a bit of a refactor. will push that later.

File: .emdaer/.offline/contributors-data.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "received_events_url": "https://api.github.com/users/flipactual/received_events",
     "type": "User",
     "site_admin": false,
-    "score": 78.48937,
+    "score": 78.615234,
```
Comment (flipactual) [inline .emdaer/.offline/contributors-data.json]: Where does this score come from/why do we track it??

**Comment by flipactual** [comment]
> I ended up creating an .offline directory underneath .emdaer to hold this and any other things like this in the future.

This should be documented for plugin creators

**Comment by flipactual** [comment]
> Currently there is no way to show a warning in emdaer, I found myself wanting to do that for the above.

I don't have an objection to adding this, but I'd want to do something like break out packages for `@emdaer/warning` and `@emdaer/error`, then add type detection to see if that's what's return, etc

We'd need to open an issue to both look into existing patterns in other projects and to determine how we want to handle it here

**Comment by infiniteluke** [comment]
I broke commitlint again

**Pull request closed without merging**